mypy/subtypes.py

```diff
@@ -228,9 +228,14 @@ def visit_typeddict_type(self, left: TypedDictType) -> bool:
         elif isinstance(right, TypedDictType):
             if not left.names_are_wider_than(right):
                 return False
-            for (_, l, r) in left.zip(right):
+            for name, l, r in left.zip(right):
                 if not is_equivalent(l, r, self.check_type_parameter):
                     return False
+                # Non-required key is not compatible with a required key since indexing
```

Comment:
Update the comment to be more specific. Here's my argument why required key shouldn't be compatible with a non-required key (assuming we had #3550 implemented):

```py
A = TypedDict('A', {'x': int})
B = TypedDict('B', {'x': int}, total=False)

def f(b: B) -> None:
    del b['x']   # Should be accepted (but is not until we implement #3550)

a: A = {'x': 0}
f(a)  # Error: if we allow this, the next line can fail
a['x']  # Should not fail with KeyError
```